reVX/wind_setbacks/wind_setbacks.py

```diff
@@ -0,0 +1,1386 @@
+
+        Returns
+        -------
+        regs : geopands.GeoDataFrame | None
+            Wind Regulations
+        Multiplier : float | None
+            Generic setbacks multiplier
+        """
+        if regs_fpath:
+            if multiplier:
+                msg = ('A wind regulation .csv file was also provided and '
+                       'will be used to determine setback multipliers!')
+                logger.warning(msg)
+                warn(msg)
+
+            multiplier = None
+            regs = self._parse_county_regs(regs_fpath, self._excl_h5)
+            logger.debug('Computing setbacks using regulations provided in: {}'
+                         .format(regs_fpath))
+        elif multiplier:
+
+            regs = None
+            if isinstance(multiplier, str):
+                multiplier = self.MULTIPLIERS[multiplier]
+
+            logger.debug('Computing setbacks using generic Max-tip Height '
+                         'Multiplier of {}'.format(multiplier))
+        else:
+            msg = ('Computing setbacks requires either a wind regulations '
+                   '.csv file or a generic multiplier!')
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        return regs, multiplier
+
+    def _rasterize_setbacks(self, shapes):
+        """
+        Convert setbacks geometries into exclusions array
+
+        Parameters
+        ----------
+        setbacks : list
+            List of (geometry, 1) pairs to rasterize. Each geometry is a
+            feature buffered by the desired setback distance in meters
+
+        Returns
+        -------
+        arr : ndarray
+            Exclusions layer arr with proper shape to write to self._excl_h5
+        """
+        logger.debug('Generating setbacks exclusion array of shape {}'
+                     .format(self._shape))
+        arr = features.rasterize(shapes=shapes,
+                                 out_shape=self._shape[1:],
+                                 fill=0,
+                                 transform=self._profile['transform'],
+                                 dtype='uint8')
+
+        return np.expand_dims(arr, axis=0)
+
+    def _general_setbacks(self, features, max_workers=None):
+        """
+        Compute general setbacks for all states either in serial or parallel
+
+        Parameters
+        ----------
+        features : list
+            list of feature files to compute setbacks from
+        max_workers : int, optional
+            Number of workers to use for setback computation, if 1 run in
+            serial, if > 1 run in parallel with that many workers, if None
+            run in parallel on all available cores, by default None
+
+        Returns
+        -------
+        setbacks : ndarray
+            Raster array of setbacks, ready to be written to the exclusions
+            .h5 file as a new exclusion layer
+        """
+        if max_workers is None:
+            max_workers = os.cpu_count()
```

Comment:
if you just do `if max_workers == 1: serial else: parallel` and send max_workers to the process pool directly with None/int you don't need to retrieve the os.cpu_count(). Obviously you can't do `max_workers > 1` if `max_workers is None` but you can do `max_workers == 1` if `max_workers is None`, Yay dynamic typing!